### PR TITLE
Add light layer on indirect lighting controller

### PR DESCRIPTION
**Pull request by EvgeniiG**
### Purpose of this PR

This PR add support of Light Layers on indirect lighting controller.

image

There is now an indirect diffuse and reflection lighting muliplier that can be affected by light layers 
Those now affect correctly all baked and screen space effect, and reflections probe and planar reflections correctly.

The previous property that was only affecting reflection probe have been keep for compatibility but don't support light layers.

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariables.hlsl [hunk 1]
```diff
@@ -279,6 +279,19 @@ float GetInversePreviousExposureMultiplier()
     return rcp(exposure + (exposure == 0.0)); // zero-div guard
 }
 
+// Helper function for indirect control volume
+float GetIndirectDiffuseMultiplier(uint renderingLayers)
+{
+    uint indirectDiffuseLayers = uint(_IndirectLightingMultiplier.y * 255.5);
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariables.hlsl:285]: Faster to reinterpret cast here.

File: com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariables.hlsl [hunk 2]
```diff
@@ -279,6 +279,19 @@ float GetInversePreviousExposureMultiplier()
...
+    return indirectDiffuseLayers & renderingLayers ? _IndirectLightingMultiplier.x : 1.0f;
+}
+
+float GetIndirectSpecularMultiplier(uint renderingLayers)
+{
+    uint indirectSpecularLayers = uint(_IndirectLightingMultiplier.w * 255.5);
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderVariables.hlsl:291]: And here.

File: com.unity.render-pipelines.high-definition/Editor/Lighting/IndirectLightingControllerEditor.cs [hunk 3]
```diff
@@ -6,21 +6,30 @@ namespace UnityEditor.Rendering.HighDefinition
...
-            PropertyField(m_IndirectSpecularIntensity, EditorGUIUtility.TrTextContent("Indirect Specular Intensity", "Sets the multiplier for reflected specular lighting."));
+            PropertyField(m_ReflectionProbeIntensityMultiplier, EditorGUIUtility.TrTextContent("Indirect Diffuse Intensity", "Sets the multiplier for baked diffuse lighting."));
+            PropertyField(m_ReflectionLightingMultiplier, EditorGUIUtility.TrTextContent("Indirect Specular Intensity", "Sets the multiplier for reflected specular lighting."));
+            PropertyField(m_ReflectionLightinglayersMask, EditorGUIUtility.TrTextContent("Indirect Specular Intensity", "Sets the multiplier for reflected specular lighting."));
+            PropertyField(m_IndirectDiffuseLightingMultiplier, EditorGUIUtility.TrTextContent("Indirect Specular Intensity", "Sets the multiplier for reflected specular lighting."));
+            PropertyField(m_IndirectDiffuseLightinglayersMask, EditorGUIUtility.TrTextContent("Indirect Specular Intensity", "Sets the multiplier for reflected specular lighting."));
```
Comment (EvgeniiG) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/IndirectLightingControllerEditor.cs:32]: Needs renaming but you are already aware.

File: com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md [hunk 4]
```diff
@@ -17,10 +17,11 @@ The **Indirect Lighting Controller** uses the [Volume](Volumes.html) framework,
 
 | Property                        | Description                                                  |
 | ------------------------------- | ------------------------------------------------------------ |
-| **Indirect Diffuse Intensity**  | A multiplier for baked and realtime Global Illumination lightmaps and Light Probes. HDRP multiplies the lightmap and Light Probe data by this value. |
-| **Indirect Specular Intensity** | A multiplier for baked, realtime, and custom Reflection Probes. HDRP multiplies the Reflection Probe data by this value. |
-
-
+| **Indirect Diffuse Lighting Multiplier**  | A multiplier for lightmaps, Light Probes, Light Probe Volumes, Screen Space Global Illumination and Raytrace Global Illumination. HDRP multiplies all those data by this value. |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md:20]: A multiplier for lightmaps, Light Probes, Light Probe Volumes, Screen-Space Global Illumination, and Ray-Traced Global Illumination. HDRP multiplies the light data from all of these by this value.

File: com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md [hunk 5]
```diff
@@ -17,10 +17,11 @@ The **Indirect Lighting Controller** uses the [Volume](Volumes.html) framework,
...
+| **Indirect Diffuse Lighting Layers** | Light layers for indirect diffuse lighting. See [LightLayers](Light-Layers.md). After you enable Light Layers, you can then use them to decouple Meshes from the this multiplier in your Scene. |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md:21]: Specifies the [Light Layers](Light-Layers.md) for indirect diffuse lighting. If you enable Light Layers, you can use them to decouple Meshes in your Scene from the above multiplier.

File: com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md [hunk 6]
```diff
@@ -17,10 +17,11 @@ The **Indirect Lighting Controller** uses the [Volume](Volumes.html) framework,
...
+| **Reflection Lighting Multiplier**  | A multiplier for baked, realtime, custom Reflection Probes and Planar Probes, Screen Space Reflection, Raytrace Reflection and Sky Reflection. HDRP multiplies all those data by this value. |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md:22]: A multiplier for baked, realtime, custom [Reflection Probes](Reflection-Probe.md) and [Planar Probes](Planar-Reflection-Probe.md), Screen-Space Reflection, Ray-Traced Reflection, and Sky Reflection. HDRP multiplies the light data from all of these by this value.

File: com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md [hunk 7]
```diff
@@ -17,10 +17,11 @@ The **Indirect Lighting Controller** uses the [Volume](Volumes.html) framework,
...
+| **Reflection Lighting Layers**  | Light layers for reflection lighting. See [LightLayers](Light-Layers.md). After you enable Light Layers, you can then use them to decouple Meshes from the this multiplier in your Scene. |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md:23]: Specifies the Light Layers for reflection lighting. If you enable Light Layers, you can use them to decouple Meshes in your Scene from the above multiplier.

File: com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md [hunk 8]
```diff
@@ -17,10 +17,11 @@ The **Indirect Lighting Controller** uses the [Volume](Volumes.html) framework,
...
+| **Reflection Probe Intensity Multiplier**  | A multiplier for baked, realtime, and custom Reflection Probes. HDRP multiplies the Reflection Probe data by this value. |
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md:24]: Is this just normal probes or both planar and normal? If it's just normal, add a link from **Reflection Probes** to Reflection-Probe.md. If it's both, use Reflection-Probes-Intro.md

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Documentation~/Override-Indirect-Lighting-Controller.md:24]: Just reflection probe, done. thanks

**Pull request merged**